### PR TITLE
Issue 19121

### PR DESCRIPTION
**Pull request by erickgonzalez**
Respect front end roles when checking if one category is the parent of another category.

Tests.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java [hunk 1]
```diff
@@ -1084,4 +1086,150 @@ private ContentType createContentTypeWithCatAndTextField(final Category parentCa
...
+                    .setKey("categoryA-1")
+                    .setCategoryVelocityVarName("categoryA-1")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA_1);
+
+            categoryAPI.save(parentCategory, childCategoryA, user, false);
+            categoryAPI.save(childCategoryA, childCategoryA_1, user, false);
+
+            Assert.assertTrue(categoryAPI.isParent(childCategoryA_1,parentCategory,user,true));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java:1136]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6106194)

File: dotCMS/src/integration-test/java/com/dotcms/rest/elasticsearch/ESContentResourcePortletTest.java [hunk 2]
```diff
@@ -184,6 +193,82 @@ public void testSearch(final TestCase testCase) throws Exception {
...
+        // Content Type with category field
+        final List<Field> fields = new ArrayList<>();
+        fields.add(new FieldDataGen().name("Title").velocityVarName("title").next());
+        fields.add(new FieldDataGen().type(CategoryField.class)
+                .name(parentCategory.getCategoryName()).velocityVarName(parentCategory.getCategoryVelocityVarName())
+                .values(parentCategory.getInode()).next());
+        final ContentType contentType = new ContentTypeDataGen().fields(fields).nextPersisted();
+
+        // Save content with grand child category
+        final Contentlet contentlet = new ContentletDataGen(contentType.id())
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rest/elasticsearch/ESContentResourcePortletTest.java:244]: Codacy found an issue: [Avoid unused local variables such as 'contentlet'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6106194)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java [hunk 3]
```diff
@@ -1084,4 +1086,150 @@ private ContentType createContentTypeWithCatAndTextField(final Category parentCa
...
+                    .setKey("categoryA-2")
+                    .setCategoryVelocityVarName("categoryA-2")
+                    .next();
+
+            categoriesToDelete.add(childCategoryA_2);
+
+            categoryAPI.save(parentCategory2, childCategoryA2, user, false);
+            categoryAPI.save(childCategoryA2, childCategoryA_2, user, false);
+
+            Assert.assertFalse(categoryAPI.isParent(childCategoryA_2,parentCategory,user,true));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java:1226]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertFalse' due to existing static import 'org.junit.Assert.assertFalse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6106194)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 import com.dotcms.IntegrationTestBase;
 import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.contenttype.business.FieldAPIImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryAPITest.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.contenttype.business.FieldAPIImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6106194)

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**